src/server/worker/pipeline/transform/registry.go

```diff
@@ -714,6 +714,12 @@ func (reg *registry) startJob(commitInfo *pfs.CommitInfo, statsCommit *pfs.Commi
 		return err
 	}
 
+	pj.ji.State = pps.JobState_JOB_RUNNING
```

Comment:
I think this is wrong (but also I think it was wrong before) - I think we should start from where a job left off in case of a master crash rather than running through the whole state machine again.  I think the `"waiting for job inputs`" above should only be run conditionally if the state is `STARTING`, but it should still advance the job to `"RUNNING"` rather than doing it here.

I don't see any problem with updating the `DataTotal` and writing out the job info here as well.